### PR TITLE
feature/461

### LINKED ISSUE #461: KSEA: Update for compliance with basic airport standards
Update KSEA to comply with the [basic airport standards](https://github.com/openscope/openscope/blob/develop/documentation/airport-file-standards.md).

### PR DESCRIPTION
**Pull request by erikquinn**
Resolves #461.

Bring KSEA into compliance with new airport specs.

### CODE REVIEW

**Comment by erikquinn** [comment]
@Fechulo @indianbhaji @n8rzz Any notes or objections?

File: assets/airports/ksea.json [hunk 1]
```diff
@@ -231,10 +315,277 @@
...
+                "34R": [["OLM", "A130|S250"], "LACEE", "ARVAD", "FOURT", "#070"]
+            },
+            "draw": [
+                ["BTG*", "COUGA", "OLM*"],
+                ["UBG*", "OLM"],
+                ["OLM", "LACEE", "ARVAD", "FOURT"]
+            ]
         }
     },
     "spawnPatterns": [
```
Comment (indianbhaji) [inline assets/airports/ksea.json]: in-game: 154acph
real life: 47.05acph

range: 47.05 - 141.15

Please reduce the acph

File: assets/airports/ksea.json [hunk 2]
```diff
@@ -280,20 +644,185 @@
...
+                ["kal/long", 0.28],
+                ["nks", 0.03],
+                ["qxe", 21.17],
+                ["skw", 10.24],
+                ["swa", 5.93],
+                ["uae", 0.41],
+                ["ual", 4.34],
+                ["vrd", 1.23]
+            ]
+        },{
```
Comment (indianbhaji) [inline assets/airports/ksea.json]: What do you want do with this? There needs to be a convention that goes throughout the file, do you want to do `}, {` or 
`},`
`{`?

File: assets/airports/ksea.json (hunk 1 above)
Comment (erikquinn) [inline assets/airports/ksea.json]: Whoa. Something's not right... i'll have a look

Comment (erikquinn) [inline assets/airports/ksea.json]: Found it. Stupid mistake.

The intended (and newly adjusted) rate is 95.25acph (54 arrivals, 41.25 departures)

File: assets/airports/ksea.json (hunk 2 above)
Comment (erikquinn) [inline assets/airports/ksea.json]: nah, this was just a mistake. 

Fixed. :+1:

File: assets/airports/ksea.json (hunk 1 above)
Comment (erikquinn) [inline assets/airports/ksea.json]: Fixed. :+1:

**Comment by erikquinn** [comment]
Bugfix merge removed, ready for review.

**Comment by erikquinn** [comment]
@Fechulo @indianbhaji bump :wink:

File: assets/airports/ksea.json [hunk 3]
```diff
@@ -61,151 +69,221 @@
         "BOANE": ["N47d59.10m0", "W122d43.52m0"],
```
Comment (Fechulo) [inline assets/airports/ksea.json]: Duplicate of BKE

File: assets/airports/ksea.json [hunk 4]
```diff
@@ -61,151 +69,221 @@
...
+        "GEG"  : ["N47d33.90m0", "W117d37.61m0"],
         "GLASR": ["N47d51.26m0", "W121d11.57m0"],
+        "GOALZ": ["N46d58.95m0", "W122d28.11m0"],
         "GRIFY": ["N47d46.15m0", "W122d24.12m0"],
         "HAROB": ["N47d14.60m0", "W123d02.45m0"],
         "HAWKZ": ["N46d47.08m0", "W122d44.34m0"],
         "HETHR": ["N47d33.54m0", "W122d08.70m0"],
         "HISKU": ["N44d30.00m0", "W122d56.65m0"],
         "HQM"  : ["N46d56.82m0", "W124d08.96m0"],
         "HQM"  : ["N46d56.82m0", "W124d08.96m0"],
```
Comment (Fechulo) [inline assets/airports/ksea.json]: Duplicate of HQM

File: assets/airports/ksea.json [hunk 5]
```diff
@@ -61,151 +69,221 @@
...
         "KMORE": ["N47d48.65m0", "W122d20.82m0"],
         "KRIEG": ["N46d19.21m0", "W122d43.33m0"],
+        "KRUZR": ["N48d04.65m0", "W120d34.68m0"],
         "KTSAP": ["N47d45.51m0", "W122d27.92m0"],
         "KWEST": ["N47d25.24m0", "W122d27.32m0"],
         "LACEE": ["N47d02.82m0", "W122d48.35m0"],
+        "LATAY": ["N46d24.47m0", "W125d04.31m0"],
         "LIINE": ["N46d50.65m0", "W122d40.11m0"],
         "LKV"  : ["N42d29.57m0", "W120d30.42m0"],
         "LKV"  : ["N42d29.57m0", "W120d30.43m0"],
```
Comment (Fechulo) [inline assets/airports/ksea.json]: Duplicate of LKV

File: assets/airports/ksea.json [hunk 6]
```diff
@@ -61,151 +69,221 @@
...
-            "ils":  [true, true],
+            "end": [
+                [47.463806, -122.317884, "415.0ft"],
+                [47.440562, -122.318092, "356.3ft"]
+            ],
+            "ils": [true, true],
             "glideslope": [3.00, 3.00]
         }
     ],
     "sids": {
```
Comment (Fechulo) [inline assets/airports/ksea.json]: The ATOME1 departure is missing.

File: assets/airports/ksea.json [hunk 7]
```diff
@@ -61,151 +69,221 @@
...
+                ["CUSBU", "WUREL"],
+                ["NESOE", "WUREL"],
+                ["DODVE", "WUREL"],
+                ["WUREL", "ATOME", "BREMM", "HAROB"],
+                ["HAROB", "HQM*"],
+                ["HAROB", "ERAVE*"],
+                ["HAROB*", "FEPOT*", "HISKU*"]
+            ]
         },
         "KMORE4": {
```
Comment (Fechulo) [inline assets/airports/ksea.json]: This procedure doesn't seem to exist anymore.

File: assets/airports/ksea.json [hunk 8]
```diff
@@ -215,11 +293,17 @@
...
             },
-            "draw": [["CUSBU", "WUREL"], ["NESOE", "WUREL"], ["DODVE", "WUREL"], ["WUREL", "ATOME", "CENEX", "KTSAP", "ZUVEN", "KMORE", "BLUIT*"]]
+            "draw": [
+                ["CUSBU", "WUREL"],
+                ["NESOE", "WUREL"],
+                ["DODVE", "WUREL"],
+                ["WUREL", "ATOME", "CENEX", "KTSAP", "ZUVEN", "KMORE*", "BLUIT*"]
+            ]
         },
         "KTSAP5": {
```
Comment (Fechulo) [inline assets/airports/ksea.json]: This procedure doesn't seem to exist anymore.

File: assets/airports/ksea.json [hunk 9]
```diff
@@ -231,10 +315,305 @@
...
+                "TOU": [["TOU", "A200+"], ["LOOOO", "A170+|S280"]],
+                "YVR": [["YVR", "A200+"], "ROESH"],
+                "YYJ": [["YYJ", "A200+"], ["ORCUS", "A170+|S280"]]
+            },
+            "body": [["MARNR", "A120+|S270"]],
+            "rwy": {
+                "16L": [["BOANE", "A110+|S250"], ["NEHOS", "A80+|S230"], ["GRIFY", "A60|S210"]],
+                "16C": [["BOANE", "A110+|S250"], ["NEHOS", "A80+|S230"], ["GRIFY", "A60|S210"]],
+                "16R": [["BOANE", "A110+|S250"], ["NEHOS", "A80+|S230"], ["GRIFY", "A60|S210"]],
+                "34L": [["PNELA", "A120+|S270"], ["VEGGN", "A150-|S250"], ["UNITT", "A100+|S230"], ["WUBET", "A80+"], ["SHIPZ", "A70|S210"], "EMMSS", "#163"],
```
Comment (Fechulo) [inline assets/airports/ksea.json]: There's a ranged restriction at VEGGN.

File: assets/airports/ksea.json (hunk 3 above)
Comment (erikquinn) [inline assets/airports/ksea.json]: fixed

File: assets/airports/ksea.json (hunk 4 above)
Comment (erikquinn) [inline assets/airports/ksea.json]: fixed

File: assets/airports/ksea.json (hunk 5 above)
Comment (erikquinn) [inline assets/airports/ksea.json]: fixed

File: assets/airports/ksea.json (hunk 6 above)
Comment (erikquinn) [inline assets/airports/ksea.json]: Wow. Went into effect 3 days ago. Neat!!!

File: assets/airports/ksea.json (hunk 7 above)
Comment (erikquinn) [inline assets/airports/ksea.json]: Must have been removed the other day when they added the ATOME1. Cool!

Removed.

File: assets/airports/ksea.json (hunk 8 above)
Comment (erikquinn) [inline assets/airports/ksea.json]: Must have been removed the other day when they added the ATOME1. Cool!

Removed.

File: assets/airports/ksea.json (hunk 9 above)
Comment (erikquinn) [inline assets/airports/ksea.json]: fixed

File: assets/airports/ksea.json (hunk 6 above)
Comment (erikquinn) [inline assets/airports/ksea.json]: Added! :smiley:

**Comment by erikquinn** [comment]
@Fechulo Hm. Worked for me, not sure about that. I'll update the branch and spin up a review app.

**Comment by erikquinn** [comment]
#32 resolved, unblocking

**Comment by erikquinn** [comment]
@n8rzz I can't make sense of these conflicts. The way you got the tests cooperating with a different airport aren't working anymore, since now the airport controller is static. I tried a slew of things to get it to work, but to no avail. When you have a chance tomorrow, could you get this sorted out for me?

**Comment by n8rzz** [comment]
@erikquinn merge conflicts are sorted out and this branch is up to date (with passing tests).  Should be good to go now.

**Pull request merged**